pkg/broker/util.go

```diff
@@ -43,6 +43,82 @@ func toBrokerPlans(apbPlans []apb.Plan) []Plan {
 	return brokerPlans
 }
 
+func updateMetadata(metadata map[string]interface{}, params []apb.ParameterDescriptor) map[string]interface{} {
+	if params == nil || len(params) == 0 {
+		return metadata
+	}
+
+	if metadata == nil {
+		metadata = make(map[string]interface{})
+	}
+
+	formDefinition := make([]interface{}, 0)
+
+	for paramIdx := 0; paramIdx < len(params); {
```

Comment:
Also, doing it as a range, because I had to alternate between the top level list and the inner list in the group was super ugly.